### PR TITLE
Java module 1.0.0 updates

### PR DESCRIPTION
**Pull request by tristan957**

### CODE REVIEW

File: mesonbuild/modules/java.py
Comment (eli-schwartz) [inline mesonbuild/modules/java.py:59]: Duplicating the type checking is unfortunate. Also, this will print the FeatureNew for `native_headers` even if you use `generate_native_headers` with a minimum meson of `>=0.63.0`.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: mesonbuild/modules/java.py
Comment (tristan957) [reply mesonbuild/modules/java.py:59]: I see. Let me move the code into a shared function then.

File: mesonbuild/modules/java.py
Comment (dcbaker) [inline mesonbuild/modules/java.py:67]:
```suggestion
        KwargInfo('package', (str, NoneType), default=None))
```

**Review by dcbaker — APPROVED** [review]

**Comment by dcbaker** [comment]
I think a rebase on master will fix the website

**Comment by tristan957** [comment]
I did rebase prior to pushing

**Comment by dcbaker** [comment]
You're right! it's an actual error in the docs :/

**Comment by tristan957** [comment]
Oh yeah. I wasn't done migrating the docs to yaml. Whoops. I didn't know what the schema was for this kind of stuff. Is there a good example or a schema I can go read?

**Comment by eli-schwartz** [comment]
https://mesonbuild.com/Yaml-RefMan.html#yaml-schema

**Comment by tristan957** [comment]
Cool. I will get to work on this hopefully before the merge window :)

**Comment by tristan957** [comment]
decided to just remove the yaml file for now. Will get to it at another time. Don't want to hold up a potential 1.0.0 release for it.

**Comment by jpakkane** [comment]
The test failures are weird. They seem unrelated but rerunning them did not make the errors go away,

```
D:\a\_temp\msys64\mingw64\lib\python3.10\site-packages\coverage\pytracer.py:290: 
CoverageWarning: Trace function changed, data is likely wrong: 
<bound method PyTracer._trace of <PyTracer at 0x1f16b29a020: 10671 lines in 109 files>> != <bound method PyTracer._trace
of <PyTracer at 0x1f164a23730: 10671 lines in 109 files>> (trace-changed)
```

**Comment by tristan957** [comment]
I don't know either

**Comment by tristan957** [comment]
Actually it looks like the same tests are failing on master?

**Comment by eli-schwartz** [comment]
Should be all fixed with #11161

**Comment by tristan957** [comment]
Cool thanks. I will rebase when yours merges

**Comment by jpakkane** [comment]
Rebasing should fix this now.

**Comment by jpakkane** [comment]
Failure is unrelated.

**Pull request merged**
